python/ql/src/semmle/python/objects/Callables.qll

```diff
@@ -333,10 +333,19 @@ class BuiltinMethodObjectInternal extends CallableObjectInternal, TBuiltinMethod
     }
 
     Builtin getReturnType() {
+        /* If we have a record of the return type in our stubs, use that. */
         exists(Builtin func |
             func = this.getBuiltin() |
             ext_rettype(func, result)
         )
+        or
+        /* Otherwise, if no such record exists, use `object` as the return type. */
+        not exists(Builtin func |
+            // We cannot do `this.getBuiltin()` here, as that would introduce negative recursion.
+            // Instead, we appeal directly to the underlying IPA type.
```

Comment:
Nitpick: was is up with the mix of multiline and singleline comments? :D

Reply:
Hey, man, I'm a free spirit. You can't tie me down to a single style of comments!

But seriously, there's no reasoning behind it. I think my preference is the `/* ... */` style, but of course I couldn't use that on lines 344-345 as autoformat would expand that into four lines. So in an effort to be forward-thinking, I used line comments there instead.